### PR TITLE
docs: add event listener guide

### PR DESCRIPTION
**Pull request by yury-s**
#5596

### CODE REVIEW

File: docs/src/events.md [hunk 1]
```diff
@@ -0,0 +1,168 @@
+---
+id: events
+title: "Events"
+---
+
+Playwright allows to listen to various types of events happening in the browser such
```
Comment (avodovnik) [inline docs/src/events.md:6]:
```suggestion
Playwright allows listening to various types of events happening in the browser, such
```

File: docs/src/events.md [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+as network requests, creation of child pages, dedicated workers etc. There are several
+ways to subscribe to such events:
+
+<!-- TOC -->
+
+## Waiting for event
+
+There is a number of methods which wait for a particular condition or an event happening
+in the page. They are usually preferred choice. Here is a few examples:
```
Comment (avodovnik) [inline docs/src/events.md:15]:
```suggestion
in the page. They are usually the preferred choice. Here are a few examples:
```

File: docs/src/events.md [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+
+Waiting for request with specified url:
```
Comment (avodovnik) [inline docs/src/events.md:17]:
```suggestion
Waiting for a request with a specified url:
```

File: docs/src/events.md [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+def print_request(request):
+  print("Request finished: " + request.url)
+page.on("requestfinished", print_request)
+page.goto("https://wikipedia.org")
+page.remove_listener("requestfinished", print_request)
+```
+
+## Adding one-off listeners
+
+If only next event needs to be handled there is convinience API for some of the events:
```
Comment (avodovnik) [inline docs/src/events.md:141]:
```suggestion
If only next event needs to be handled, there is a convenience API for some of the events:
```

**Review by avodovnik — CHANGES_REQUESTED** [review]

File: docs/src/events.md [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+```
+
+```python sync
+with page.expect_popup() as popup:
+  page.evaluate("window.open()")
+popup.value.goto("https://wikipedia.org")
+```
+
+## Adding event listener
+
```
Comment (aslushnikov) [inline docs/src/events.md:82]: I'd add a small description to preface the code snippet, along these lines:

Playwright supports traditional language mechanisms to subscribe and unsubscribe from events

File: docs/src/events.md [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+```
+
+```python sync
+def print_request(request):
+  print("Request finished: " + request.url)
+page.on("requestfinished", print_request)
+page.goto("https://wikipedia.org")
+```
+
+## Removing event listener
```
Comment (aslushnikov) [inline docs/src/events.md:107]: merge this with `adding event listener`? 2x less snippets, less bloat?

**Review by aslushnikov — APPROVED** [review]

File: docs/src/events.md (hunk 5 above)
Comment (yury-s) [reply docs/src/events.md:82]: Done.

File: docs/src/events.md (hunk 6 above)
Comment (yury-s) [reply docs/src/events.md:107]: Done.

**Review by aslushnikov — APPROVED** [review]

**Pull request merged**